In [5]:
print(type(True))

<class 'bool'>


In [ ]:
# typecasting

gpa=3.2

print(type(gpa))

print(type(float(gpa)))

<class 'float'>
<class 'int'>


In [ ]:
# input

name=input("What is your name?: ")
age=int(input('How old are you?'))
print(f'hello {name}, you are {age} years old')

hello anshul, you are 23 years old


In [16]:
# Calc Area of Rectangle

def areaOfRectangle(length,breadth):
    return (length*breadth)

l = float(input("Enter the length of Rectangle"))
b = float(input("Enter the breadth of Rectangle"))

print(f'Area of Rectangle is {areaOfRectangle(l,b)} units squared')

Area of Rectangle is 15.75 units squared


In [17]:
""" Madlibs game
word game where you create a story 
by filling in blanks with random words
"""

adjective1 = input('Enter an adjective')
noun1 = input('Enter a noun')
adjective2 = input('Enter another adjective')
verb1 = input('Enter a verb ending with \'ing\'')
adjective3 = input('Enter another adjective')

print(f'Today I went to a {adjective1} zoo.')
print(f'I saw an {noun1}')
print(f'{noun1} was {adjective2} and {verb1}.')
print(f'I was {adjective3}!')

Today I went to a large zoo.
I saw an Parrot
Parrot was Ferocious and Worrying.
I was thrilled!


In [ ]:
import math

x=9.1

print(math.pi)
print(math.e)
print(math.sqrt(x))
print(math.ceil(x)) 
print(math.floor(x))

3.141592653589793
2.718281828459045
3.0166206257996713
10
9


In [ ]:
 Calculator

num1 = (input('Enter first operand'))
num2 = (input('Enter second operand'))

op = int(input('Choose the operation:\n1) Addition\n2) Subtraction\n3) Multiplication\n4) Division'))
result=None
try:
    if op==1:
        result = num1+num2
    elif op==2:
        result = num1-num2
    elif op==3:
        result = num1*num2
    elif op==4:
        result = num1/num2
    else:
        print("Enter a valid operation from Menu!")
except Exception as e:
    print(f'The exception {e} surfaced!')


print(f'The Result is {result}')

The exception can't multiply sequence by non-int of type 'str' surfaced!
The Result is None


In [29]:
 odd or even

num = 52

print("odd" if num%2==1 else "even")  ternary operator

even


In [ ]:
name = input("Enter your name!")

str1 = "Lola"
str2 = "Amrikano"

 result = len(name)
 result=name.strip(" ")
 result = name.split()
 joined = " ".join(result)
 join2 = " ".join([str1, str2])   join expects an iterable, so use a list

 has O(n)
result = name.find("a") # first occurence
result = name.rfind("a") # last occurence

# capitalise string (first name only)
capitalised = name.capitalize()
capitalised = name.upper() # lower also
capitalised = name.casefold() # more rigorous than lower, german B is translated to ss.
# capitalised = name.title()
print(capitalised)

print("Contains Only Digits" if name.isdigit() else "Does not contain only Digits")

print("Contains Only Alphanumeric" if name.isalnum() else "Does not Contain Only Alphanumeric")

# print(result)
# print(joined)
# print(join2)


anshul
Does not contain only Digits
Contains Only Alphanumeric


In [45]:
phone_number = input("Enter your phone number")

# print("Number is irish" if phone_number.startswith("+353") else "Number is not irish")

# count dashes in No
DashesNo=phone_number.count('-')
print(DashesNo)

spacedNo=phone_number.replace('-',' ')
print(spacedNo)

onlyDigits = "".join(c for c in phone_number if c.isdigit())
print(onlyDigits)

2
+353 89 2128452
353892128452


In [46]:
help(str)

Help on class str in module builtins:

class str(object)
 |  str(object='') -> str
 |  str(bytes_or_buffer[, encoding[, errors]]) -> str
 |
 |  Create a new string object from the given object. If encoding or
 |  errors is specified, then the object must expose a data buffer
 |  that will be decoded using the given encoding and error handler.
 |  Otherwise, returns the result of object.__str__() (if defined)
 |  or repr(object).
 |  encoding defaults to sys.getdefaultencoding().
 |  errors defaults to 'strict'.
 |
 |  Methods defined here:
 |
 |  __add__(self, value, /)
 |      Return self+value.
 |
 |  __contains__(self, key, /)
 |      Return bool(key in self).
 |
 |  __eq__(self, value, /)
 |      Return self==value.
 |
 |  __format__(self, format_spec, /)
 |      Return a formatted version of the string as described by format_spec.
 |
 |  __ge__(self, value, /)
 |      Return self>=value.
 |
 |  __getattribute__(self, name, /)
 |      Return getattr(self, name).
 |
 |  __getitem__(

In [ ]:
# Validate user input exercise 
# 1st username is no more than 12 characters 
# second username must not contain spaces 
# 3rd username must not contain digits

username=input('Enter your username')

if len(username)>12:
    print("Too Long! Enter up to 12 characters!")
elif username.find(" ")!=-1:
    print("Username contains Spaces! No Spaces Allowed!")
# elif not username.isalpha():
#     print("Username Must not Contain Digits!")
elif any(c.isdigit() for c in username):
    print("Username Must not Contain Digits!")

Username contains Spaces! No Spaces Allowed!


: 

In [77]:
# Indexing
# Accessing elements of a sequence using [] (indexing operators)
# [start; end; step]

credit_number = "1234-5678-9012-3456"

print(credit_number[5])
print(credit_number[4:8])
print(credit_number[4:])
print(credit_number[:3])
print(credit_number[-1])
print(credit_number[-3:])
print(credit_number[::5])

last_Digits = credit_number[-4:]

print(f"XXXX-XXXX-XXXX-{last_Digits}")


5
-567
-5678-9012-3456
123
6
456
1593
XXXX-XXXX-XXXX-3456


# format specifiers = {value:flags} format a value based on what # flags are inserted

.(number)f = round to that many decimal places (fixed point) 

:(number) = allocate that many spaces

:03 = allocate and zero pad that many spaces

:< = left justify

:> = right justify

:^ = center align

:+ = use a plus sign to indicate positive value

:= = place sign to leftmost position

: = insert a space before positive numbers

:, = comma separator



In [15]:
price1 = 300000000.14159
price2 = -98700.65
price3 = 12000.34

print(f'Price 1 is ${price1:+,.2f}')
print(f'Price 2 is ${price2:,}')
print(f'Price 3 is ${price3:,}')

Price 1 is $+300,000,000.14
Price 2 is $-98,700.65
Price 3 is $12,000.34


In [17]:
# while loops

name = input("Enter your name")

while name=="":
    print("You did not enter your name!")
    name = input("Enter your name")
print(f'Hello {name}')

You did not enter your name!
You did not enter your name!
You did not enter your name!
Hello askdasd


In [18]:
food = input('Enter your favourite food (q to quit).')

while food!='q':
    print(f'Your favourite food is {food}')
    food = input('Enter your favourite food (q to quit).')
print('app closed')

Your favourite food is pumpkin
Your favourite food is tomato
Your favourite food is onion
Your favourite food is chicken
app closed


In [21]:
num = int(input('Enter a number between 1 to 10'))

while num < 1 or num > 10:
    print('Number is not between 1 to 10')
    num = int(input('Enter a number between 1 to 10'))
print(f'Your entered number is {num}')


Number is not between 1 to 10
Number is not between 1 to 10
Number is not between 1 to 10
Number is not between 1 to 10
Number is not between 1 to 10
Your entered number is 5


In [24]:
# Compound interest Calculator

principal = float(input('Enter your Principal'))
rate_of_interest = float(input('Enter your rate_of_interest'))
time = float(input('Enter the time'))

total_amt = principal*pow((1+rate_of_interest/100),time)

print(f'The total amount returned is {total_amt:.2f}')

The total amount returned is 572.45


# For Loops

In [27]:
# for x in reversed(range(1,11)):
for x in range(1,11,3):    
    print(x)

print("Happy New Year!")

1
4
7
10
Happy New Year!


countdown timer in python

In [37]:
import time

# time.sleep(3)

# print("Time's Up!")

timer = int(input("Timer for how many seconds?"))

for x in reversed(range(1, timer+1)):
    
    seconds = x % 60
    minutes = (x//60) % 60
    hours = x // 3600
    print(f'{hours:02}:{minutes:02}:{seconds:02}', end='\r')
    time.sleep(1)

print("The Time's Up!")

The Time's Up!


# Nested Loops

In [38]:
for x in range(1,10):
    print(x, end="")

123456789

| `end` value | Output example for numbers 1–3               |
| ----------- | -------------------------------------------- |
| `'\n'`      | 1\n2\n3 (default, prints each on a new line) |
| `' '`       | 1 2 3 (space-separated)                      |
| `','`       | 1,2,3 (comma-separated)                      |
| `'-'`       | 1-2-3                                        |
| `''`        | 123 (no separator)                           |
| `'\t'`      | 1␉ 2␉ 3 (tab-separated)                      |
| `'***'`     | 1***2***3                                    |


In [41]:
# Make a Rectangle of Symbols

rows = int(input('Enter # of Rows'))
columns = int(input('Enter # of Columns'))
symbol = input('Enter the symbol')

for r in range(rows):
    for c in range(columns):
        print(symbol, end="")
    print()

iiii
iiii
iiii
